src/Umbraco.Web.UI.Client/src/views/propertyeditors/datepicker/datepicker.controller.js

```diff
@@ -87,13 +147,25 @@ function dateTimePickerController($scope, notificationsService, assetsService, a
 			    var element = $element.find("div:first");
 
 				// Open the datepicker and add a changeDate eventlistener
-			    element
-			        .datetimepicker(angular.extend({ useCurrent: true }, $scope.model.config))
-			        .on("dp.change", applyDate)
-			        .on("dp.error", function(a, b, c) {
-			            $scope.hasDatetimePickerValue = false;
-			            $scope.datePickerForm.datepicker.$setValidity("pickerError", false);
-			        });
+			    element.datetimepicker(angular.extend({
+                    useCurrent: true,
+                    minDate: moment($scope.model.config.minDate).isValid() ? $scope.model.config.minDate : moment({ y: 1900 }),
+                    maxDate: moment($scope.model.config.maxDate).isValid() ? $scope.model.config.maxDate : moment().add(100, 'y')
+			    }, $scope.model.config))
+			    .on("dp.change", applyDate)
+			    .on("dp.error", function (a, b, c) {
+			        $scope.hasDatetimePickerValue = false;
+			        $scope.datePickerForm.datepicker.$setValidity("pickerError", false);
+			    });
+
+                // Ensure table table cell with class "picker-switch" fill 5 cols, when not using calendarWeeks.
+			    if ($scope.model.config.calendarWeeks == false) {
+			        var currentDateInput = element.find(".datepickerinput"),
+                        dateInputs = $("input[name=datepicker].datepickerinput"),
+                        datePickerIndex = dateInputs.index(currentDateInput);
+
+			        $(".bootstrap-datetimepicker-widget").eq(datePickerIndex).find("table > thead .picker-switch").attr("colspan", 5);
+			    }

```

Comment:
This part just change colspan from 6 to 5, when "calendarWeeks" isn't used, so the datepicker use full width of the dialog.